api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/processor/ChildDateOfBirthCalculator.java

```diff
@@ -0,0 +1,78 @@
+        this.cycleEntitlementCalculator = cycleEntitlementCalculator;
+    }
+
+    /**
+     * Calculates how many of the children's dates of birth listed on the given PaymentCycle have birthdays
+     * which would affect the next payment after this cycle. The dates that would affect the next payment
+     * are between the last entitlement date of the current cycle and the final entitlement date of the next
+     * cycle. Those with birthdays on the start boundary date (the date of the final entitlement date of
+     * the current cycle) are not included, but those on the end boundary date (the final entitlement date
+     * of the next cycle) are included.
+     *
+     * @param paymentCycle The current PaymentCycle including the relevant dates of birth of the children.
+     * @return The number of children who will soon turn 4 that affect the next Payment.
+     */
+    public int getNumberOfChildrenTurningFourWhichAffectsNextPayment(PaymentCycle paymentCycle) {
+        return getNumberOfChildrenTurningAgeWhichAffectsNextPayment(paymentCycle, 4);
+    }
+
+    /**
+     * As per the method above but for children turning 1.
+     *
+     * @param paymentCycle The current PaymentCycle including the relevant dates of birth of the children.
+     * @return The number of children who will soon turn 1 that affect the next Payment.
+     */
+    public int getNumberOfChildrenTurningOneWhichAffectsNextPayment(PaymentCycle paymentCycle) {
+        return getNumberOfChildrenTurningAgeWhichAffectsNextPayment(paymentCycle, 1);
+    }
+
+    private int getNumberOfChildrenTurningAgeWhichAffectsNextPayment(PaymentCycle paymentCycle, int age) {
+        if (CollectionUtils.isEmpty(paymentCycle.getChildrenDob())) {
+            return 0;
+        }
+        LocalDate currentCycleStartDate = paymentCycle.getCycleStartDate();
+        LocalDate nextCycleStartDate = paymentCycle.getCycleEndDate().plusDays(1);
+        LocalDate lastEntitlementDateInCurrentCycle = getLatestEntitlementDateFromCycleStartDate(currentCycleStartDate);
+        LocalDate lastEntitlementDateInNextCycle = getLatestEntitlementDateFromCycleStartDate(nextCycleStartDate);
+        return Math.toIntExact(paymentCycle.getChildrenDob().stream()
+                .filter(childDob -> isWithinPeriodExcludingStartBoundaryIncludingEndBoundary(
+                        lastEntitlementDateInCurrentCycle,
+                        lastEntitlementDateInNextCycle,
+                        age,
+                        childDob))
+                .count());
+    }
+
+    private LocalDate getLatestEntitlementDateFromCycleStartDate(LocalDate cycleStartDate) {
+        return cycleEntitlementCalculator.getVoucherEntitlementDatesFromStartDate(cycleStartDate)
+                .stream().max(LocalDate::compareTo).get();
+    }
+
+    private boolean isWithinPeriodExcludingStartBoundaryIncludingEndBoundary(LocalDate lastEntitlementDateInCurrentCycle,
```

Comment:
Suggest 'Date' instead of 'Boundary' for easier readability

Reply:
Yep, updated.